From 9abe7ccf2708ae25b1a242b67a46c1b5cf97e785 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 19 Jul 2022 14:49:49 +0800
Subject: [PATCH 25/62] Revert "drm/etnaviv: check for reaped mapping in
 etnaviv_iommu_unmap_gem"

This reverts commit 436cff507f2a41230baacc3e2ef1d3b2d2653f40.
---
 drivers/gpu/drm/etnaviv/etnaviv_mmu.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/gpu/drm/etnaviv/etnaviv_mmu.c b/drivers/gpu/drm/etnaviv/etnaviv_mmu.c
index 9ba2fe48228f..984569a59a90 100644
--- a/drivers/gpu/drm/etnaviv/etnaviv_mmu.c
+++ b/drivers/gpu/drm/etnaviv/etnaviv_mmu.c
@@ -282,12 +282,6 @@ void etnaviv_iommu_unmap_gem(struct etnaviv_iommu_context *context,
 
 	mutex_lock(&context->lock);
 
-	/* Bail if the mapping has been reaped by another thread */
-	if (!mapping->context) {
-		mutex_unlock(&context->lock);
-		return;
-	}
-
 	/* If the vram node is on the mm, unmap and remove the node */
 	if (mapping->vram_node.mm == &context->mm)
 		etnaviv_iommu_remove_mapping(context, mapping);
-- 
2.27.0

